src/book/mod.rs

```diff
@@ -367,9 +367,13 @@ impl MDBook {
         self
     }
 
-    pub fn test(&mut self) -> Result<(), Box<Error>> {
+    pub fn test(&mut self, library_paths: Vec<&str>) -> Result<(), Box<Error>> {
         // read in the chapters
         self.parse_summary()?;
+        let library_args: Vec<&str> = (0..library_paths.len()).map(|_| "-L")
+                                                              .zip(library_paths.into_iter())
+                                                              .flat_map(|x| vec![x.0, x.1])
```

Comment:
It's a little annoying that `tuple` does not implement `IntoIterator` trait, causing this `vec![x.0, x.1]` nonsense.